kubescape/host-scanner
# test: add unit test improvements

## Overview
Here are some unit-tests to improve the code coverage of the `host-scanner`.
We started from a **26.6%** of coverage and now we got a **51.6%**, so an increment of **25%**.
In order to have a better coverage, we should refactor some functions, especially the ones that make use of system capabilities (like read/write from file system), separating the read/write from the resto of the main function and providing the content read as input for the new functions.
Here's an example:
* https://endler.dev/2018/go-io-testing/
* https://stackoverflow.com/questions/56397240/unit-test-function-that-is-opening-and-reading-a-file

## How to Test
To run the test with the coverage generation:
```
CGO_ENABLED=0 go test -coverprofile=coverage.out -cover -v ./...
```
To generate and html with the result:
```
go tool cover -html=coverage.out -o coverage.html
```
You can open the `coverage.html` file in your browser and see the visual result.

## Checklist before requesting a review
